resources/js/components/ui/InfoIcon.tsx

```diff
@@ -0,0 +1,50 @@
+import {A, AnchorProps} from "@solidjs/router";
+import {ImInfo} from "solid-icons/im";
+import {Match, Switch, VoidComponent} from "solid-js";
+import {htmlAttributes, useLangFunc} from "../utils";
+import {Button} from "./Button";
+
+interface ButtonProps extends htmlAttributes.button {
+  href?: undefined;
+}
+
+interface LinkProps extends AnchorProps {
+  href: string;
+}
+
+type Props = ButtonProps | LinkProps;
+
+/**
+ * A tiny blue (i) icon providing more information to a control it is next to.
+ * It can be a button, or an internal or external link.
+ */
+export const InfoIcon: VoidComponent<Props> = (props) => {
+  const t = useLangFunc();
+  const icon = <ImInfo class="inlineIcon !mb-0.5 text-blue-500" size="16" />;
+  return (
+    <Switch>
+      <Match when={props.href && props}>
```

Comment:
props zawsze będzie truthy, więc tu nie ma sensu go dodawać. Nawet jak nie będzie propsów, to props będzie wynosił {}, więc !!props zawsze === true.

Reply:
Ale brzydko tam macie, że wartość when potem gdzieś trafia :p

Reply:
Props jest oczywiście zawsze truthy, to jest po to żeby dostać poprawnie zawężony typ w funkcji przekazanej jako children. Akurat i tak na koniec jest potrzebny cast ale są sytuacje  gdzie to wystarcza żeby on rozumiał że propsy są już discriminated.